spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/otlp/OtlpProperties.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2012-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.autoconfigure.tracing.otlp;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * Configuration properties for {@link OtlpAutoConfiguration}.
+ *
+ * @author Jonatan Ivanov
+ * @since 3.1.0
+ */
+@ConfigurationProperties("management.otlp.tracing")
```

Comment:
Hi @jonatan-ivanov , I see that spring is diverging from the configuration properties proposed by the otel-sdk. For instance, the properties managed by this class maps to otel-sdk "otel.exporter.otlp.endpoint", "otel.exporter.otlp.compression" and so on...

Are there any plans to converge the config property names to the same used by otel-sdk ?

If no, isnt there another way to make them at least seem easier to map. E.g, spring version would only add the management.otlp prefix ?

Reply:
Here's the rationale behind the current structure: https://github.com/spring-projects/spring-boot/issues/30381 This PR follows that pattern.

There are at least two tricky things here:
1. There are at least two components that can export OTLP right now in Boot, the Micrometer Tracing OTel bridge (in this PR) and Micrometer's OTLP Meter Registry. I think users need to be able to configure them separately and `otel.exporter.otlp.endpoint` does not really make that distinction.
1. Under the hood, shipping OTLP does not need to use the OTel SDK at all, e.g.: Micrometer's OTLP registry does not use it, if for example Brave will add OTLP support, I guess that won't either. Check out the Zipkin properties and auto-configuration. Boot does not use OTel properties there and it needs to auto-configure either OTel or Brave based on those properties.

I think right now this is the easiest thing you can do to set these:
```
otel.exporter.otlp.xyz=...
management.otlp.metrics.export.xyz=${otel.exporter.otlp.xyz}
management.otlp.tracing.xyz=${otel.exporter.otlp.xyz} // maybe I should add export here?
```

Reply:
I think we should figure out what would be the best and your feedback is apprechiated, right now this is what we have:
- [`OtlpProperties`](https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/otlp/OtlpProperties.java) for metrics
  - has a different prefix: `management.otlp.metrics.export` (this PR has `management.otlp.tracing`)
  - has a different property `url` (this PR calls it `endpoint`)
- [`ZipkinProperties`](https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/zipkin/ZipkinProperties.java) for tracing (Brave/OTel)
  - has similar prefix: `management.zipkin.tracing` (this PR follows that convention)
- [OTel SDK Properties](https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md#exporters) 
  - has the prefix: `otel.exporter.otlp`
  - has similar properties

Reply:
> 1. There are at least two components that can export OTLP right now in Boot, the Micrometer Tracing OTel bridge (in this PR) and Micrometer's OTLP Meter Registry. I think users need to be able to configure them separately and `otel.exporter.otlp.endpoint` does not really make that distinction.

Fair enough, makes sense as users could possibly use 2 different otel-compatible backends: one for metrics and other for tracing

> management.otlp.metrics.export.xyz=${otel.exporter.otlp.xyz}

Im doing the opposite: trying to support the SDK props as they are more widely known, so end-users would be able to configure it based on SDK docs, but it is really hard to map all the spring props to the sdk props. If it was just a matter of prefixes, life would be easier. But I understand the complexity and the other scenarios that you mention. There's even a more complex one such as `management.tracing.sampling.probability=${otel.traces.sampler.arg}` - the `otel.traces.sampler.arg` (urgh!) is a beast on its own as it includes inner properties on it depending on the sampler implementation. Perhaps somewhere in the docs we could have a section mapping the spring->otel-sdk properties.

> * has a different prefix: `management.otlp.metrics.export` (this PR has `management.otlp.tracing`)
> 
> * has a different property `url` (this PR calls it `endpoint`)

I personally like `management.otlp.tracing.export.http.endpoint` more. 
I also liked that you exposed all the OtlpHttpSpanExporterBuilder configurations, different from the `OpenTelemetryAutoConfiguration.otelSpanProcessor` which is using the BatchSpanProcessor internally and not offering the possibility to configure its propertyes such as batch/queue size, timeouts and so on...